concent_api/core/models.py

```diff
@@ -73,3 +75,216 @@ class MessageAuth(Model):
+        Failed                  = 'failed'
+
+    class Meta:
+        unique_together = (
+            ('requestor', 'task_id'),
+            ('requestor', 'subtask_id'),
+        )
+
+    task_id                     = models.CharField(max_length = MESSAGE_TASK_ID_MAX_LENGTH)
+    subtask_id                  = models.CharField(max_length = MESSAGE_TASK_ID_MAX_LENGTH)
+
+    # Relation to concent client in requestor context.
+    requestor                   = models.ForeignKey(Client, related_name = 'subtasks_as_requestor')
+
+    # Relation to concent client in provider context.
+    provider                    = models.ForeignKey(Client, related_name = 'subtasks_as_provider')
+
+    state                       = models.CharField(max_length = 32, choices = SubtaskState.choices())
+
+    # If in an active state, it's the time at which Concent automatically transitions to a passive state if it does
+    # not get the information it expects from the client (a timeout). Must be NULL in a passive state.
+    next_deadline               = models.DateTimeField(null = True)
+
+    # Related messages
+    task_to_compute             = models.ForeignKey(StoredMessage, null = True)
+    report_computed_task        = models.ForeignKey(StoredMessage, null = True)
+    ack_report_computed_task    = models.ForeignKey(StoredMessage, null = True)
+    reject_report_computed_task = models.ForeignKey(StoredMessage, null = True)
+    subtask_results_accepted    = models.ForeignKey(StoredMessage, null = True)
+    subtask_results_rejected    = models.ForeignKey(StoredMessage, null = True)
+
+
+class ReceiveQueue(models.Model):
+    """
+    Stores information about messages to be returned from the `receive` or `receive-out-of-band` endpoint.
+    """
+
+    class ConcentResponseType(ChoiceEnum):
+        ForceReportComputedTask         = 'ForceReportComputedTask'
+        ForceReportComputedTaskResponse = 'ForceReportComputedTaskResponse'
+        VerdictReportComputedTask       = 'VerdictReportComputedTask'
+        ForceGetTaskResultRejected      = 'ForceGetTaskResultRejected'
+        ForceGetTaskResultFailed        = 'ForceGetTaskResultFailed'
+        ForceGetTaskResultUpload        = 'ForceGetTaskResultUpload'
+        ForgetGetTaskResultDownload     = 'ForgetGetTaskResultDownload'
+        ForceSubtaskResults             = 'ForceSubtaskResults'
+        SubtaskResultsSettled           = 'SubtaskResultsSettled'
+        ForceSubtaskResultsResponse     = 'ForceSubtaskResultsResponse'
+        SubtaskResultsRejected          = 'SubtaskResultsRejected'
+
+    class ReceiveQueueType(ChoiceEnum):
+        Receive             = 'receive'
+        ReceiveOutOfBand    = 'receive_out_of_band'
+
+    response_type        = models.CharField(max_length = 32, choices = ConcentResponseType.choices())
+    client               = models.ForeignKey(Client)
+    queue                = models.CharField(max_length = 32, choices = ReceiveQueueType.choices())
```

Comment:
Same as with state. Add constants for length and assert that all values have length smaller than that.